spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/BatchingSqsClientAdapter.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.operations;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import org.springframework.util.Assert;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+import software.amazon.awssdk.services.sqs.batchmanager.SqsAsyncBatchManager;
+import software.amazon.awssdk.services.sqs.model.*;
+
+/**
+ * An {@link SqsAsyncClient} adapter that provides automatic batching capabilities using AWS SDK's
+ * {@link SqsAsyncBatchManager}.
+ * 
+ * <p>
+ * This adapter automatically batches SQS operations to improve performance and reduce costs by combining multiple
+ * requests into fewer AWS API calls. All standard SQS operations are supported: send message, receive message, delete
+ * message, and change message visibility.
+ * 
+ * <p>
+ * <strong>Important - False Positives Warning:</strong> This adapter processes requests asynchronously through
+ * batching. Method calls may return successfully before the actual request is sent to AWS SQS. This can result in false
+ * positives where the operation appears to succeed locally but fails during the actual transmission to AWS.
+ * Applications should:
+ * <ul>
+ * <li>Always handle the returned {@link CompletableFuture} to detect actual transmission errors</li>
+ * <li>Implement appropriate error handling and monitoring</li>
+ * <li>Consider retry mechanisms for critical operations</li>
+ * </ul>
+ * 
+ * <p>
+ * <strong>Batch Optimization:</strong> The underlying {@code SqsAsyncBatchManager} from the AWS SDK bypasses batching
+ * for {@code receiveMessage} calls that include per-request configurations for certain parameters. To ensure batching
+ * is not bypassed, it is recommended to configure these settings globally on the {@code SqsAsyncBatchManager} builder
+ * instead of on each {@code ReceiveMessageRequest}. The parameters that trigger this bypass are:
+ * <ul>
+ * <li>{@code messageAttributeNames}</li>
+ * <li>{@code messageSystemAttributeNames}</li>
+ * <li>{@code messageSystemAttributeNamesWithStrings}</li>
+ * <li>{@code overrideConfiguration}</li>
+ * </ul>
+ * <p>
+ * By configuring these globally on the manager, you ensure consistent batching performance. If you require per-request
+ * attribute configurations, using the standard {@code SqsAsyncClient} without this adapter may be more appropriate.
+ * @author Heechul Kang
+ * @since 3.2
```

Comment:
This will likely be part of the upcoming 4.0.0 release. @maciejwalkowiak, let me know if you have different thoughts.